src/main/java/com/maddyhome/idea/vim/newapi/IjVimSearchGroup.kt

```diff
@@ -125,6 +125,19 @@ open class IjVimSearchGroup : VimSearchGroupBase(), PersistentStateComponent<Ele
     showSearchHighlight = injector.globalOptions().hlsearch
   }
 
+  override fun isSomeTextHighlighted(): Boolean {
+    val vimEditors = injector.editorGroup.getEditors().filter {
+      (injector.application.isUnitTest() || it.ij.component.isShowing)
+    }
+    for (vimEditor in vimEditors) {
+      val editor = vimEditor.ij
+      if (editor.vimLastHighlighters != null) {
+        return true
+      }
+    }
+    return false
+  }
+
```

Comment:
This is fine, but personally, I'd prefer an implementation like I outlined in [this discussion comment](https://github.com/JetBrains/ideavim/discussions/1036#discussioncomment-11253104) - implement `v:hlsearch` as a variable with state in `VimVariableServiceBase` and move state from `VimSearchGroup` to the variable service. 

The `v:hlsearch` variable would then be the source of truth of whether or not highlights were active, and there wouldn't be a need to check with all editors or introduce a new API.